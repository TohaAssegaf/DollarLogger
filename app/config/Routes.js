export const HOME = 'HOME_ROUTE'

export const UPDATE_BUDGET = 'UPDATE_BUDGET_ROUTE'

export const LOADING = 'LOADING_ROUTE'