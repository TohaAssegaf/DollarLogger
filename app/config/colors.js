export const PRIMARY_BUTTON_COLOR = "#6A1B9A"

export const HEADER_BACKGROUND_COLOR = "#6A1B9A"
export const HEADER_TEXT_COLOR = "#FFFFFF"

export const THIN_BORDER_COLOR = "#222222"