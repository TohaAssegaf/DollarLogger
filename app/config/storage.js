// Path for budget in async storage.
export const BUDGET_ASYNC_STORAGE_KEY = '@DollarLogger:budget'
