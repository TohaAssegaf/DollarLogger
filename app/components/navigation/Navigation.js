import * as Routes from '/app/config/Routes'
import Home from '/app/screens/home/Home'
import Loading from '/app/screens/loading/Loading'
import AddPayment from '/app/screens/payment/AddPayment'
import UpdateBudget from '/app/screens/updatebudget/UpdateBudget'
import React from 'react'
import {
  StackNavigator
} from 'react-navigation'

export default StackNavigator({
  [Routes.UPDATE_BUDGET]: {
    screen: UpdateBudget,
    path: Routes.UPDATE_BUDGET
  },
  [Routes.HOME]: {
    screen: Home,
    path: Routes.HOME
  },
  [Routes.LOADING]: {
    screen: Loading,
    path: Routes.LOADING
  },
  [Routes.ADD_PAYMENT]: {
    screen: AddPayment,
    path: Routes.ADD_PAYMENT
  }
}, {
  initialRouteName: Routes.LOADING
})
