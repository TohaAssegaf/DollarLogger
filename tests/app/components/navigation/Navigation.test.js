import * as Routes from '~/app/config/Routes'
import Navigation from '~/app/components/navigation/Navigation'
import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

const mockStore = configureMockStore([thunk])

it('renders correctly', () => {
  const store = mockStore({ budget: { isFetchComplete: false }})
  const rendered = renderer.create(<Provider store={store}><Navigation /></Provider>).toJSON()
  expect(rendered).toMatchSnapshot();
});
