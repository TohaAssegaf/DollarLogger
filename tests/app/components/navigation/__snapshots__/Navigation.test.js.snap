// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#EFEFF4",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#222222",
                "fontSize": 30,
              }
            }
          >
            What is your weekly budget?
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 15,
                "paddingBottom": 20,
              }
            }
          >
            (Don't worry, you can change this later)
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#222222",
                "borderBottomWidth": 0.5,
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "fontSize": 30,
                  },
                ]
              }
            >
              $
            </Text>
            <TextInput
              allowFontScaling={true}
              keyboardType="numeric"
              onBlur={[Function]}
              onChangeText={[Function]}
              style={
                Array [
                  Object {
                    "fontSize": 20,
                    "minWidth": 100,
                  },
                  Object {
                    "fontSize": 30,
                  },
                ]
              }
              value={undefined}
            />
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingTop": 40,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "paddingHorizontal": 40,
                }
              }
            >
              <View
                accessibilityComponentType="button"
                accessibilityLabel="Update weekly budget"
                accessibilityTraits={
                  Array [
                    "button",
                  ]
                }
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={undefined}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "padding": 8,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "#6A1B9A",
                        },
                      ]
                    }
                  >
                    SUBMIT
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
